liquibase-core/src/main/java/liquibase/command/AbstractDatabaseCommandStep.java

```diff
@@ -0,0 +1,92 @@
+package liquibase.command;
+
+import liquibase.GlobalConfiguration;
+import liquibase.Scope;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.integration.commandline.CommandLineUtils;
+import liquibase.integration.commandline.LiquibaseCommandLineConfiguration;
+import liquibase.resource.ResourceAccessor;
+
+import java.util.ResourceBundle;
+
+import static java.util.ResourceBundle.getBundle;
+
+/**
+ * Convenience base class for {@link CommandStep} implementations which need a Database object.
+ */
+public abstract class AbstractDatabaseCommandStep extends AbstractCommandStep {
```

Comment:
Using inheritance like this to manage the database/connection in commands is not the architectural direction we are trying to move to. The plan is to leverage the "pipeline" structure we built into the command framework. 

So while this is helpful in the short term, I don't like adding it to our public API for everyone to use. 

At a minimum, it would probably be good to mark it as "deprecated, this class is going to be removed in a future version" so 3rd party people can know to avoid it. We may even want to make it package protected or even moved out of liquibase-core to avoid the temptation?

Reply:
I will remove the class.